services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/statistics/StatisticsService.java

```diff
@@ -172,6 +180,52 @@ private void gatherFlowStats(IOFSwitch iofSwitch) {
         }
     }
 
+    @NewCorrelationContextRequired
+    private void gatherFlowStatsPerTable(IOFSwitch iofSwitch) {
+        OFFactory factory = iofSwitch.getOFFactory();
+
+        OFTableStatsRequest flowStatsRequest = factory
+                .buildTableStatsRequest()
+                .build();
+
+        tableStatsLogger.trace("Getting flow stats per table for switch={}", iofSwitch.getId());
+
+        Futures.addCallback(iofSwitch.writeStatsRequest(flowStatsRequest),
+                new FutureCallback<List<OFTableStatsReply>>() {
+
+                @Override
+                public void onSuccess(@Nullable List<OFTableStatsReply> result) {
+                    if (CollectionUtils.isEmpty(result)) {
+                        return;
+                    }
+
+                    result.stream()
+                            .filter(reply -> CollectionUtils.isNotEmpty(reply.getEntries()))
+                            .map(OFTableStatsReply::getEntries)
+                            .flatMap(List::stream)
+                            .forEach(entry -> {
+                                if (entry.getLookupCount().equals(entry.getMatchedCount())) {
+                                    tableStatsLogger.debug(
+                                            "Lookup and matched counters are equal ({}) on the switch {} table {}",
+                                            entry.getLookupCount().getValue(), iofSwitch.getId(),
+                                            entry.getTableId().getValue());
+                                } else {
+                                    tableStatsLogger.warn(
+                                            "Lookup {} and matched {} counters are not equal on the switch {} table {}",
+                                            entry.getLookupCount().getValue(), entry.getMatchedCount().getValue(),
+                                            iofSwitch.getId(), entry.getTableId().getValue());
+                                }
+                            });
+                }
+
+                @Override
+                public void onFailure(Throwable t) {
+                    tableStatsLogger.error("Failed to read table stats", t);
```

Comment:
I think we should add  SwitchId information

 ` on switch={}", iofSwitch.getId()`